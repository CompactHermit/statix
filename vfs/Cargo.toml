[package]
name = "vfs"
version = "0.0.0"
edition = "2018"
license = "MIT"

[dependencies]
indexmap = "1.6.2"

