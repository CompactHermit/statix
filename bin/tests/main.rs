mod util {
    #[macro_export]
    macro_rules! test_lint {
        ($($tname:ident),*,) => {
            test_lint!($($tname),*);
        };
        ($($tname:ident),*) => {
            $(
                #[test]
                fn $tname() {
                    use statix::{config::OutFormat, traits::WriteDiagnostic, lint};
                    use vfs::ReadOnlyVfs;

                    let file_path = concat!("data/", stringify!($tname), ".nix");
                    let contents = include_str!(concat!("data/", stringify!($tname), ".nix"));

                    let vfs = ReadOnlyVfs::singleton(file_path, contents.as_bytes());

                    let mut buffer = Vec::new();
                    vfs.iter().map(lint::lint).for_each(|r| {
                        buffer.write(&r, &vfs, OutFormat::StdErr).unwrap();
                    });

                    let stripped = strip_ansi_escapes::strip(&buffer).unwrap();
                    let out =  std::str::from_utf8(&stripped).unwrap();
                    insta::assert_snapshot!(&out);
                }
            )*
        };
    }
}

test_lint! {
    bool_comparison,
    empty_let_in,
    manual_inherit,
    manual_inherit_from,
    legacy_let_syntax,
    collapsible_let_in,
    eta_reduction,
    useless_parens,
    unquoted_splices,
    empty_pattern,
    redundant_pattern_bind,
    unquoted_uri,
    deprecated_is_null,
}
