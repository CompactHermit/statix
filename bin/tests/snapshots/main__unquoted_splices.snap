---
source: bin/tests/main.rs
expression: "&out"

---
[W09] Warning: Found unquoted splice expression
   ╭─[data/unquoted_splices.nix:7:3]
   │
 7 │   ${x}
   ·   ──┬─  
   ·     ╰─── Consider quoting this splice expression
───╯
[W09] Warning: Found unquoted splice expression
   ╭─[data/unquoted_splices.nix:8:3]
   │
 8 │   ${toString (x + y)}
   ·   ─────────┬─────────  
   ·            ╰─────────── Consider quoting this splice expression
───╯
[W09] Warning: Found unquoted splice expression
   ╭─[data/unquoted_splices.nix:9:5]
   │
 9 │   a.${toString x}
   ·     ──────┬──────  
   ·           ╰──────── Consider quoting this splice expression
───╯
[W09] Warning: Found unquoted splice expression
    ╭─[data/unquoted_splices.nix:12:3]
    │
 12 │ ╭─▶   ${
 14 │ ├─▶   }
    · │         
    · ╰───────── Consider quoting this splice expression
────╯

